pkg/controller/pod.go

```diff
@@ -604,6 +604,38 @@ func (c *Controller) handleAddPod(key string) error {
 		}
 		if isVmPod && c.config.EnableKeepVmIP {
 			pod.Annotations[fmt.Sprintf(util.VmTemplate, podNet.ProviderName)] = vmName
+			ipName := ovs.PodNameToPortName(vmName, namespace, podNet.ProviderName)
+			ipCr, err := c.config.KubeOvnClient.KubeovnV1().IPs().Get(context.Background(), ipName, metav1.GetOptions{})
+			if err != nil {
+				if !k8serrors.IsNotFound(err) {
+					errMsg := fmt.Errorf("failed to get ip CR %s: %v", ipName, err)
+					klog.Error(errMsg)
+					return errMsg
+				}
+				// the returned pointer is not nil if the CR does not exist
+				ipCr = nil
+			}
+			if ipCr != nil {
+				if ipCr.Spec.Subnet != subnet.Name {
+					key := fmt.Sprintf("%s/%s", pod.Namespace, vmName)
+					ports, err := c.ovnClient.ListNormalLogicalSwitchPorts(true, map[string]string{"pod": key})
+					if err != nil {
+						klog.Errorf("failed to list lsps of pod '%s', %v", pod.Name, err)
+						return err
+					}
+					for _, port := range ports {
+						if err != nil {
+							klog.Warningf("failed to get port '%s' sg, %v", port.Name, err)
+						}
+						// when lsp is deleted, the port of pod is deleted from any port-group automatically.
+						klog.Infof("gc logical switch port %s", port.Name)
+						if err := c.ovnLegacyClient.DeleteLogicalSwitchPort(port.Name); err != nil {
+							klog.Errorf("failed to delete lsp %s, %v", port.Name, err)
+							return err
+						}
+					}
+				}
+			}
```

Comment:
Is it possible to abstract the process as a function and called in  handleAddPod?

Reply:
> Is it possible to abstract the process as a function and called in handleAddPod?
it sounds good

Reply:
> Is it possible to abstract the process as a function and called in handleAddPod?

it sounds ok，i will change it